src/VisualStudio/Core/SolutionExplorerShim/DiagnosticItem/DiagnosticItem.cs

```diff
@@ -154,15 +148,34 @@ private void UpdateRuleSetFile(string pathToRuleSet, ReportDiagnostic value)
             var analyzerID = _analyzerItem.AnalyzerReference.Display;
             var rules = FindOrCreateRulesElement(ruleSetDocument, analyzerID);
             var rule = FindOrCreateRuleElement(rules, _descriptor.Id);
-            rule.Attribute("Action").Value = newAction;
+
+            if (value == ReportDiagnostic.Default)
+            {
+                // If the new severity is 'Default' we just delete the entry for the rule from the ruleset file.
+                // In the absence of an explicit entry in the ruleset file, the rule reverts back to its 'Default' severity.
```

Comment:
@tmeschter I meant to convey that the rule reverts back to its default severity so far as the 'current' ruleset goes. I will update the comment.

Do I need to handle the base ruleset case somehow? We don't handle that currently do we?